src/test/java/plugins/MSBuildPluginTest.java

```diff
@@ -0,0 +1,133 @@
+package plugins;
+
+import static org.junit.Assert.assertThat;
+
+import java.io.IOException;
+
+import javax.inject.Named;
+
+import org.hamcrest.Matchers;
+import org.jenkinsci.test.acceptance.junit.AbstractJUnitTest;
+import org.jenkinsci.test.acceptance.junit.TestActivation;
+import org.jenkinsci.test.acceptance.junit.WithOS;
+import org.jenkinsci.test.acceptance.junit.WithPlugins;
+import org.jenkinsci.test.acceptance.msbuild.MSBuildInstallation;
+import org.jenkinsci.test.acceptance.msbuild.MSBuildStep;
+import org.jenkinsci.test.acceptance.po.Build;
+import org.jenkinsci.test.acceptance.po.FreeStyleJob;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.internal.AssumptionViolatedException;
+
+import com.google.inject.Inject;
+
+/**
+ * This test is designed to be run on a Windows machine with MSBuild installed
+ * 
+ * In order for the test to know where the MSBuild installation dir is, an
+ * environment variable must be set. i.e:
+ * 
+ *      MSBUILD_EXE = C:\Windows\Microsoft.NET\Framework\v2.0.50727\MSBuild.exe
+ * 
+ * Tests will be skipped if:
+ *      - The environment variable is not set 
+ *      - The test is not running on a Windows machine
+ *      - MSBUild.exe is not present in the specified location
+ */
+@WithPlugins("msbuild")
+@TestActivation({"MSBUILD_EXE"})
+@WithOS(os = {WithOS.OS.WINDOWS})
+public class MSBuildPluginTest extends AbstractJUnitTest {
+
+    @Inject
+    @Named("MSBuildPluginTest.MSBUILD_EXE") 
+    private String MS_BUILD_EXECUTABLE;
```

Comment:
IMHO and IIUC we face here the same issue as in `@WithOS`. We would be searching the PATH env variable for the machine running the tests, not _necessarily_ the machine running Jenkins. Actually, same concern is present ATM for the `@Native` annotation.

Just to be sure I get this right anyway, the idea here would be to scan all directories present in the path looking for files named as the value passed to the `@Native` annotation (i.e "msbuild" or "java", correct?  

I am happy to implement the solution you think it's best given your expertise on the matter (plus I will need your bee at some point :)), so could you please let me know what you prefer (`@WithOS` or tweaking `@Native` annotation)? and then I will proceed with the implementation.